tools/checklocks/state.go

```diff
@@ -24,12 +24,22 @@ import (
 	"golang.org/x/tools/go/ssa"
 )
 
+// lockType represents the current state of the lock.
+// The 0 value represents the unlocked state.
+type lockType int
+
+const (
+	unlocked lockType = iota
+	locked
+	// TODO(jamesyou): DEFERUNLOCK
+)
+
 // lockState tracks the locking state and aliases.
 type lockState struct {
 	// lockedMutexes is used to track which mutexes in a given struct are
 	// currently locked. Note that most of the heavy lifting is done by
 	// valueAsString below, which maps to specific structure fields, etc.
-	lockedMutexes []string
+	lockedMutexes map[string]lockType
```

Comment:
What is confusing at the moment is that we have a map from mutexes to their lock states. State is a set that can contain `unlocked`. But we still call this lockedMutexes. Does just "mutexes" make more sense here now?